test/header_parser_tests/macros.h

```diff
@@ -0,0 +1,11 @@
+#define TEST1 1.1
+#define TEST2 10
+#define TEST3 (TEST1 + TEST2)
+#define TEST4 "test"
+
+// This should have been ignored but is parsed as an int of value 1.
```

Comment:
I would not expect this to be ignored, I would expect this to be evaluated as a 'comma expression'.

```c++
#define TEST5 4, \
              2, \
              3

int bla3(){
    return TEST5; // returns 3.
}
```

Please check support for comma expressions and support for `\` in expressions separately.

And if this is a bug in llvm/clang, file an issue there, and an issue here pointing to that issue.

Reply:
Just tested this out,
```c
int a = bla3(); // this returns 3 as expected.
int b = TEST5; // this however doesn't compile and libclang returns the first value in the comma sequence.
```

To make this compile I tried this
```c
#define TEST5 1,2,3

int bla3(){
    return TEST5; // returns 3.
}
int main()
{
    int a = bla3();
    int b;
    b = TEST5; // this compiles now
    printf("%d %d\n", a, b);

    return 0;
}
```
GCC Output: `3 1`

So I think this comma operator is behaves differently incase of functions or variables.